sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,301 @@
+
+
+def test_pin_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    pin_joint = PinJoint('pin_joint', parent, child, child_point_pos=(0, l, 0))
+
+    assert pin_joint.name == 'pin_joint'
+    assert pin_joint.parent == parent
+    assert pin_joint.child == child
+    assert pin_joint._parent_joint_location == Vector(0)
+    assert pin_joint._child_joint_location == l * child.frame.y
+    assert pin_joint.parent_axis == parent.frame.x
+    assert pin_joint.child_axis == child.frame.x
+
+    theta = dynamicsymbols('pin_joint_theta')
+    thetad = dynamicsymbols('pin_joint_theta', 1)
+    omega = dynamicsymbols('pin_joint_omega')
+    assert pin_joint.coordinates == [theta]
+    assert pin_joint.speeds == [omega]
+    assert pin_joint.kds == [thetad - omega]
+
+    child_joint_point = pin_joint.child_joint_point
+    parent_joint_point = pin_joint.parent_joint_point
+    assert parent_joint_point.name == 'pin_joint_parent_joint'
+    assert child_joint_point.name == 'pin_joint_child_joint'
+    assert child_joint_point.pos_from(parent_joint_point) == Vector(0)
+    assert child_joint_point.pos_from(child.masscenter) == l * child.frame.y
+    assert parent_joint_point.pos_from(parent.masscenter) == Vector(0)
+    assert child.masscenter.pos_from(parent.masscenter) == - l * child.frame.y
+    assert child.frame.ang_vel_in(parent.frame) == omega * parent.frame.x
+
+
+def test_pin_joint_arguments():
+    parent = Body('parent')
+    child = Body('child')
+    l1 = Symbol('l1')
+    l2 = Symbol('l2')
+    pin_joint = PinJoint('pin_joint', parent, child,
+                         parent_point_pos=(0, l1, 0),
+                         child_point_pos=(0, l2, 0),
+                         parent_axis=parent.frame.y,
+                         child_axis=child.frame.y)
+    assert pin_joint.name == 'pin_joint'
+    assert pin_joint.parent == parent
+    assert pin_joint.child == child
+    assert pin_joint.parent_axis == parent.frame.y
+    assert pin_joint.child_axis == child.frame.y
+    assert pin_joint._parent_joint_location == l1 * parent.frame.y
+    assert pin_joint._child_joint_location == l2 * child.frame.y
+
+
+def test_sliding_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    sliding_joint = SlidingJoint('sliding_joint', parent, child,
+                                 child_point_pos=(l, 0, 0))
+
+    assert sliding_joint.name == 'sliding_joint'
+    assert sliding_joint.parent == parent
+    assert sliding_joint.child == child
+    assert sliding_joint._parent_joint_location == Vector(0)
+    assert sliding_joint._child_joint_location == l * child.frame.x
+    assert sliding_joint.parent_axis == parent.frame.x
+    assert sliding_joint.child_axis == child.frame.x
+
+    dis = dynamicsymbols('sliding_joint_dis')
+    disd = dynamicsymbols('sliding_joint_dis', 1)
+    vel = dynamicsymbols('sliding_joint_vel')
```

Comment:
In the rotational joints you are using mathematical variables (theta, omega, etc), but in these joints you are using "dis" and "vel". You should be consistent, either use mathematical variable or not. Also isn't position preferred over distance for this? For three linear coordinates you can make use of `xyz` and `uvw` or `x_x`, `x_y`, `x_z`, `v_x`, `v_y`, `v_z`.